pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -746,7 +746,12 @@ func buildGatewayNetworkFiltersFromTCPRoutes(node *model.Proxy, push *model.Push
 		// based on the match port/server port and the gateway name
 		for _, tcp := range vsvc.Tcp {
 			if l4MultiMatch(tcp.Match, server, gateway) {
-				return buildOutboundNetworkFilters(node, tcp.Route, push, port, v.Meta)
+				out := make([]*listener.Filter, 0)
+				if server.GetTls().GetMode() == networking.ServerTLSSettings_ISTIO_MUTUAL {
+					out = append(out,
+						buildMetadataExchangeNetworkFilters(istionetworking.ListenerClassTCPIstioMTLSGateway)...)
```

Comment:
As tls mode is checked out of the function, does the new type needed? 

Asking this because the type can be passed to many functions.

Reply:
Sorry for the late reply. I am very very strongly against a new ListenerClass. Adding new enum types is one of the biggest source of bugs in Istio historically.

If you don't want to change all callsites, just make a new function `buildMetadataExchangeNetworkFiltersForMTLSGateway` or something